File: 1431. Kids With the Greatest Number of Candies/main.py
from typing import List


class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        maxCandies = max(candies)
        result = []

        for current in candies:
            result.append(current + extraCandies >= maxCandies)

        return result

File: 1431. Kids With the Greatest Number of Candies/main.rs
struct Solution;

impl Solution {
    pub fn kids_with_candies(candies: Vec<i32>, extra_candies: i32) -> Vec<bool> {
        let max_candies = *candies.iter().max().unwrap();
        let mut result = Vec::new();

        for current in candies.into_iter() {
            result.push(current + extra_candies >= max_candies);
        }

        result
    }
}

impl Solution {
    pub fn kids_with_candies(candies: Vec<i32>, extra_candies: i32) -> Vec<bool> {
        let max_candies = *candies.iter().max().unwrap();

        candies
            .iter()
            .map(|current| current + extra_candies >= max_candies)
            .collect()
    }
}
